modules/ROOT/pages/tuning-repeatable-nonrepeatable.adoc

```diff
@@ -0,0 +1,84 @@
+= Repeatable vs. Non-Repeatable Streaming
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Understanding streaming strategies and properly establishing which one applies best depending on your use case, is key to get the most of your flows in terms of performance.
+
+To determine the streaming strategy that best suits your use case, first identify if you need:
+
+* A non-repeatable stream (read payload only once) or
+* A repeatable stream (read payload more than once)
+
+== Non-Repeatable Streaming
+
+If you can handle your payload as a non-repeatable stream, disable this feature by enabling the following attribute in the connector:
+
+[source,xml,linenums]
+----
+<file:read path="exampleFile.json">
+  <non-repeatable-stream />
+</file:read>
+----
+
+== Repeatable Streaming
+Alternatively, if your use case needs to read the payload multiple times or concurrently, you can choose between:
+
+* file-store streaming strategies or
+* in-memory streaming strategies
+
+Adjusting the in-memory buffers size (defaults to 512KB) based on your payload size and the expected concurrency.
+
+=== File-Store Repeatable Stream
+
+The following example sets a specific buffer size for file-store repeatable stream, if you need to read a file that is always about 1 MB, you can configure a 1 MB buffer:
+
+[source,xml,linenums]
+----
+<file:read path="bigFile.json">
+  <repeatable-file-store-stream
+    inMemorySize="1"
+    bufferUnit="MB"/>
+</file:read>
+----
+
+ The previous values were provided as an example. You should always run performance tests to find the optimal buffer size for your use case.
+
+==== File-Store Buffer Size
+For file storage streaming, to handle larger or smaller files, you can change the buffer size `inMemorySize` to optimize performance. There are some options: +
+
+* Configure a larger buffer size to avoid the number of times Mule runtime engine needs to write the buffer on disk. This increases performance, but it also limits the number of concurrent requests your application can process because it  requires more memory.
+* Configure a smaller buffer size to save memory load at the expense of response time.
+
+
```

Comment:
Please apply the suggestions made to the previous peer topic to this one as well.

Reply:
Edited for simple changes below, in addition to ^